### PR TITLE
Add a `note: ` prefix to all notes.

### PR DESCRIPTION
**Pull request by zygoloid**
Omit the `note: ` prefix and the snippet from the "in import" note that precedes a diagnostic.

This makes our diagnostic output more closely match that of Clang and GCC.

### CODE REVIEW

File: toolchain/diagnostics/diagnostic_consumer.cpp [hunk 1]
```diff
@@ -27,10 +27,15 @@ auto StreamDiagnosticConsumer::HandleDiagnostic(Diagnostic diagnostic) -> void {
...
+        *stream_ << "note: ";
+        break;
+      case DiagnosticLevel::Context:
         break;
     }
     *stream_ << message.format_fn(message) << "\n";
-    message.loc.FormatSnippet(*stream_);
+    if (message.level != DiagnosticLevel::Context) {
+      message.loc.FormatSnippet(*stream_);
+    }
```
Comment (chandlerc) [inline toolchain/diagnostics/diagnostic_consumer.cpp]: I'm OK starting here, but I think we should add a TODO or do some follow-up work as a consequence.

At the moment, the `in import` message doesn't contain the name of the imported or importing library or any other information. The name of the imported library however is visible in the snippet before this change and removed after. So it somewhat looks like a regression.

However, I don't think relying on the snippet for this information is the right call, so I do think this is the right direction. My guess is we want to revisit the contextual diagnostics to tweak their output at some point, but happy for that to be in the future.

**Review by chandlerc — APPROVED** [review]
LGTM as I think this is a good initial step, but see inline -- I think it exposes some additional work we should probably make a note of on the import diagnostics. With the snippets, it wasn't as obviously needed, but seems more important now.

File: toolchain/diagnostics/diagnostic.h [hunk 2]
```diff
@@ -18,8 +18,12 @@
 namespace Carbon {
 
 enum class DiagnosticLevel : int8_t {
-  // A note, not indicating an error on its own, but possibly providing context
-  // for an error.
+  // Information about the location of another diagnostic, showing how we
+  // reached that location. This is currently only used for the "in import"
+  // message.
+  Context,
```
Comment (jonmeow) [inline toolchain/diagnostics/diagnostic.h]: We have a lot of things named "Context" in the toolchain, and the uses of this look like `, Context,`. What do you think of "PrefixNote", "Origin", or something like that?

File: toolchain/diagnostics/diagnostic_consumer.cpp (hunk 1 above)
Comment (jonmeow) [reply toolchain/diagnostics/diagnostic_consumer.cpp]: FWIW, my take would be that while the message could provide a library name, shouldn't we still associate a line of code when possible? i.e., if a developer is trying to figure out why they're getting the library, wouldn't that be helpful?

Comment (chandlerc) [reply toolchain/diagnostics/diagnostic_consumer.cpp]: I think the location is still there, as part of the message prefix? It's just the snippet being elided.

Comment (jonmeow) [reply toolchain/diagnostics/diagnostic_consumer.cpp]: Sure then, the snippet. I find snippets useful, not sure why we'd elide them. If this is what you both want though, can you please add comments explaining the choice?

Comment (zygoloid) [reply toolchain/diagnostics/diagnostic_consumer.cpp]: Comment added.

File: toolchain/diagnostics/diagnostic.h (hunk 2 above)
Comment (zygoloid) [reply toolchain/diagnostics/diagnostic.h]: Renamed to `LocationInfo`, does that work better for you?

File: toolchain/diagnostics/diagnostic_consumer.cpp (hunk 1 above)
Comment (chandlerc) [reply toolchain/diagnostics/diagnostic_consumer.cpp]: FWIW, I don't feel strongly about the snippets either way. I somewhat see both sides -- they both seem useful and a bit verbose. I guess matching Clang and GCC tilts me very, very slightly towards eliding them, but I think it'd be fine to revisit that if this is a big regression Jon. Or revisit it by exploring other ways to capture some of the value.

But maybe that's OK to fix-forward given the `note: ` churn that this PR already has?

**Comment by chandlerc** [comment]
(All of this still LG, but good to wait for Jon before landing)

File: toolchain/diagnostics/diagnostic.h (hunk 2 above)
Comment (jonmeow) [reply toolchain/diagnostics/diagnostic.h]: Seems good enough.

File: toolchain/diagnostics/diagnostic_consumer.cpp (hunk 1 above)
Comment (jonmeow) [reply toolchain/diagnostics/diagnostic_consumer.cpp]: I'm in disagreement, but I think I'm in the minority.

**Review by jonmeow — APPROVED** [review]

**Pull request merged**